solr/core/src/java/org/apache/solr/cloud/RecoveryStrategy.java

```diff
@@ -188,6 +189,9 @@ public final void close() {
     close = true;
     cancelPrepRecoveryCmd();
     log.warn("Stopping recovery for core=[{}] coreNodeName=[{}]", coreName, coreZkNodeName);
+    if (replicationHandlerDoingFetch != null) {
```

Comment:
[Q] Technically, there's a possibility here that 'replicationHandlerDoingFetch' is unset after we've checked the conditional, but before we call `abortFetch()1.  If that happens, it's lead to a NPE on the `abortFetch()` call.  But it seems so remote that I'm not sure it's worth the complexity of guarding against it....